src/runtime.cc

```diff
@@ -311,15 +327,14 @@ namespace detail {
 		// traffic will occur, and `runtime` can stop functioning as a scheduler_delegate (which would require m_exec to be live).
 		m_exec.reset();
 
+		// task_manager references the scheduler as its delegate, so we destroy it first.
+		m_task_mngr.reset();
+
 		// ~executor() joins its thread after notifying the scheduler that the shutdown epoch has been reached, which means that this notification is
 		// sequenced-before the destructor return, and `runtime` can now stop functioning as an executor_delegate (which would require m_schdlr to be live).
-		// m_schdlr references the task instances managed m_task_mngr, so we destroy it first. m_task_mngr uses m_schdlr as a delegate, but does not call to the
-		// delegate from its destructor.
+		// m_schdlr references the task instances managed m_task_mngr, so we destroy it first.
```

Comment:
This looks outdated, the task instances are managed by `task_graph`, and this is not being destroyed first?

Reply:
You're right!